Warning cleanup.

barisaydinoglu opened the pull request:
- Warnings cleaned:
  - {c} is a raw type. References to generic type {c}<{t}> should be parameterized.
  - The serializable class {c} does not declare a static final serialVersionUID field of type long.
  - The static field {c}.{t} should be accessed in a static way.
  - Type safety: Unchecked cast from Object to {t}.
  - Unnecessary @SuppressWarnings("{w}").
- TimeSpanTest moved to **src/main/java** to **src/test/java**, and rewritten as JUnit test.

serious6 commented on src/main/java/microsoft/exchange/webservices/data/IAsyncResult.java:
```diff
@@ -16,7 +16,7 @@
  */
 
-public interface IAsyncResult extends Future {
+public interface IAsyncResult extends Future<Object> {
```
perhabs this is useful: [Java Generics - Unbounded Wildcards](http://docs.oracle.com/javase/tutorial/java/generics/unboundedWildcards.html)

vboctor replied:
Thanks @serious6.  That is pretty much what I expected, Object matches everything except simple types.

serious6 commented on src/main/java/microsoft/exchange/webservices/data/AbstractAsyncCallback.java:
```diff
@@ -12,14 +12,14 @@
 
-abstract class AbstractAsyncCallback implements Runnable, Callback {
-  Future task;
+abstract class AbstractAsyncCallback implements Runnable, Callback<Object> {
```
same issue: `Callback<Object>` -> `Callback<?>`

serious6 commented:
perhabs we should have a closer look at the choosen generics as you mentioned by @vboctor´s line-comment. @barisaydinoglu I think some of the choosen generics are made to wide. for example:
```Java
public interface IAsyncResult extends Future<Object> {
  public Object getAsyncState();
  public WaitHandle getAsyncWaitHanle();
  public boolean getCompleteSynchronously();
  public boolean getIsCompleted();
}
```
I think it should be (using raw types)
```Java
public interface IAsyncResult extends Future<T> {
  public T getAsyncState();
  public WaitHandle getAsyncWaitHanle();
  public boolean getCompleteSynchronously();
  public boolean getIsCompleted();
}
```
perhabs this could be applied to more classes?

vboctor commented:
As @serious6 mentioned in his comment with example about use of generics. I think we should use the following model:

- When accepting a parameter, we can use the most generic type that meets our needs (e.g. List<Object> or List<?>).
- When returning a type, or deriving from a type, I would rather use T (e.g. List<T>).  This enables code interacting with this type to get the full richness of T.

Now we can always use T to simplify the above (model mostly used in C#).  That will give the full power of generics with some extra generated code for each of the specific types used.

barisaydinoglu commented on src/main/java/microsoft/exchange/webservices/data/EwsUtilities.java:
```diff
@@ -1035,35 +1036,28 @@ protected static TimeSpan getXSDurationToTimeSpanValue(String xsDuration) {
-      year = Integer.parseInt(m.group().substring(0,
-          m.group().indexOf("Y")));
-    }
+    //    m = Pattern.compile("(\\d+)Y").matcher(xsDuration);
```
the parameter "year" is not used.

barisaydinoglu commented on src/main/java/microsoft/exchange/webservices/data/EwsXmlReader.java:
```diff
@@ -422,7 +422,7 @@ public String readValue() throws XMLStreamException,
     } else if (this.presentEvent.getEventType() == XmlNodeType.CHARACTERS
         && this.presentEvent.isCharacters()) {
                         /*
-			 * if(this.presentEvent.asCharacters().getData().equals("<")) {
```
There are some errors with project formatter.
- Comment line formatting:
 - IntelliJ: does not format comment line indentation. So, every indentation of the code above is right based on your formatter.
 - Eclipse: formats (or changes) comment line indentation.
- google-styleguide for Java works different for Eclipse and Intellij.
- Mix of space and tab are being used as comment indentation character.
 - A tab could be a different number of columns depending on your environment, but a space is always one column.
 - In terms of how many spaces (or tabs) constitutes indentation, it's more important to be consistent throughout your code than to use any specific tab stop value.

serious6 replied:
@barisaydinoglu I also noticed this kind of gap in the styleguide and also realized that the provided formatters of the google-site have it´s differences. Running these against checkstyle also shows that there is something wrong with it and it is just bad that this topic keeps chasing us. Perhabs we can have a seperate PR with the things you noticed changed in the codeStyleSettings.xml. I opened #158 for this. Nevertheless the files with untouched code should be reverted in my opinion.

barisaydinoglu commented:
I've completed changes below:
- Reverted reindented lines which don't include any code changes.
- Cleaned ```@SuppressWarnings```.
- Inserted empty lines before fields.
- Corrected comments of ```serialVersionUID```.

serious6 commented:
_merged with ref. 5aeb2e7aaddb89138959f02b1a1d86fa796c0405_
Thanks for the contribution
